### PR TITLE
feat: experimental health frontend

### LINKED ISSUE #9938: Use health check to check Kubernetes cluster connectivity
### Is your enhancement related to a problem? Please describe

See https://github.com/podman-desktop/podman-desktop/issues/9818

### Describe the solution you'd like

Use the healthcheck endpoint(s) to check cluster connectivity

### Describe alternatives you've considered

_No response_

### Additional context

There is currently no function to check  the health of a cluster in the Kubernetes library, but seems easy to implement, and there is the opportunity to provide a patch for the library: https://github.com/kubernetes-client/javascript/issues/693

### TODO

- [x] backend part: #10076 
- [ ] dispatch to frontend: https://github.com/podman-desktop/podman-desktop/pull/10349
- [ ] frontend part: https://github.com/podman-desktop/podman-desktop/pull/10349

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

Display the health of kubernetes contexts in Settings > Kubernetes

### What issues does this PR fix or reference?

Fixes #9938 

### How to test this PR?

Activate the experimental mode in `~/.local/share/containers/podman-desktop/configuration/settings.json`:

```
"kubernetes.statesExperimental": true
```
- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by axel7083** [comment]
> Display the health of kubernetes contexts in Settings > Kubernetes

I guess you are talking about the following ?

image

**Review by axel7083 — APPROVED** [review]
LGTM

File: packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte [hunk 1]
```diff
@@ -125,7 +158,7 @@ async function handleDeleteContext(contextName: string) {
         <div class="grow flex-column divide-gray-900 text-[var(--pd-invert-content-card-text)]">
           <div class="flex flex-row">
             <div class="flex-none w-36">
-              {#if $kubernetesContextsState.get(context.name)?.reachable}
+              {#if isContextReachable(context.name, experimentalStates)}
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte]: I would say, that from UI side (html part), we should not call or do any computation at the display time, so no call to a map or a function, the data providing already all the conditions for the rendering.

so IMHO it should be part of the data we're receiving, so only

```svelte
{#if someObjectToDisplay.reachable}
   ....
{/if}
```svelte

and not utility function called at various places in the .svelte/html part

File: packages/renderer/src/stores/kubernetes-context-health.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export const kubernetesContextsHealths = readable<ContextHealth[]>([], set => {
+  window.events?.receive('kubernetes-contexts-healths', (value: unknown) => {
+    set(value as ContextHealth[]);
+  });
+  window
+    .kubernetesGetContextsHealths()
+    .then(value => set(value))
+    .catch((err: unknown) => console.log('Error getting contexts healths', err));
+});
```
Comment (benoitf) [inline packages/renderer/src/stores/kubernetes-context-health.ts]: question: why this store is not using the new EventStore pattern so we can monitor and track the calls ?

Comment (feloy) [reply packages/renderer/src/stores/kubernetes-context-health.ts]: I considered using it, but I hardly measure the cost of using it.

I'll look more deeply at it

File: packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte (hunk 1 above)
Comment (feloy) [reply packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte]: I moved the logic out of the html. Is it what you expect, or do we want to create a store for this?

File: packages/renderer/src/stores/kubernetes-context-health.ts (hunk 2 above)
Comment (feloy) [reply packages/renderer/src/stores/kubernetes-context-health.ts]: ok, I replaced with an EventStore

**Review by benoitf — APPROVED** [review]
thanks, the enhancement in the svelte part is much more readable (on my end)

thanks for enhancing the store to use the EventStore pattern

**Pull request merged**